src/algorithms/masked_gibbs.jl

```diff
@@ -510,3 +510,50 @@ function assert_spikes_not_in_mask(
         end
     end
 end
+
+function clean_masks(
+    masks::Vector{Mask},
+    num_neurons::Integer
+)
+    new_masks = Mask[]
+    for n = 1:num_neurons
+        # Find the masks corresponding to thie neuron
+        this_neur_masks = Mask[]
+        for mask = 1:length(masks)
+            if masks[mask][1] == n
+                push!(this_neur_masks, masks[mask])
+            end
+        end
+
+        # Find the start of mask times for this neuron
+        start_times = []
+        for (n,(t0,t1)) in this_neur_masks
+            push!(start_times, t0)
+        end
+
+        # Sort them
+        start_times_indices = sortperm(start_times)
+        sorted_mask = this_neur_masks[start_times_indices]
+
+        # Find the offending masks
+        indices_to_delete = []
+        for index = 1:(length(sorted_mask)-1)
+            if (sorted_mask[index+1][2][1] - sorted_mask[index][2][2] < 0.0001)
+                push!(indices_to_delete, index)
+            end
+        end
+
+        # Go through and remove them
+        counter = 0
+        for index = 1:length(sorted_mask)
+            if index in indices_to_delete
+                counter = counter + 1
+            else
+                push!(new_masks, (sorted_mask[index][1],(sorted_mask[index-counter][2][1],sorted_mask[index][2][2])))
+                counter = 0
+            end
+        end
```

Comment:
I'm having a little trouble following this. Could you make the comment a bit more clear?

```
# Remove them sequentially, in each case merging them into the mask occurring first. 
```

Does this code appropriately handle the situation where you have three (or more) masks that occur contiguously? They all end up getting merged into one? It is easy for me to see that this works for the case where it is just two continguous masks.

Reply:
It does handle the case where there are many masks all overlapping (for example a set of masks 0-2, 2-4 and 4-6 become one mask 0-6). 

The way it does this is through the `counter` variable lying around: it counts how many consecutive masks there are that overlap, then once you bump into the first new non-contiguous mask we enter the else statement. The push! in the else statement adds a 'clean mask' that starts at the start time of the mask `counter` steps back in the list (i.e. the last one that was non-contiguous) and finished at the time of the last contiguous mask.

It's hard to say in words, but is simpler than that garbled paragraph suggests; hopefully it makes sense though?

Reply:
Ah yes, that makes perfect sense now. How about something like this below? I think it'll help me remember what is going on.

```julia
# Remove them sequentially, in each case merging them into the mask occurring last. 
for index = 1:length(sorted_mask)
    if index in indices_to_delete
       # Delete mask (i.e. don't push it onto new_masks array).
        num_deleted = num_deleted + 1

    else
        # Keep mask, modify start time to reflect any deleted masks.
        neuron_id = sorted_mask[index][1]
        start = sorted_mask[index - num_deleted][2][1]
        stop = sorted_mask[index][2][2]
        push!(new_masks, (neuron_id, (start, stop)))

        # Reset the number of deleted masks.
        num_deleted = 0
    end
end
```

Reply:
Yup, that is clearer, will put it in